core/src/test/java/org/elasticsearch/ingest/PipelineFactoryTests.java

```diff
@@ -86,6 +86,38 @@ public void testCreateWithPipelineOnFailure() throws Exception {
         assertThat(pipeline.getOnFailureProcessors().get(0).getType(), equalTo("test-processor"));
     }
 
+    public void testCreateWithPipelineEmptyOnFailure() throws Exception {
```

Comment:
These tests will now, right? Because of the empty on_failure processors? I think we should expect an exception here instead.